nodestream/pipeline/extractors/stores/aws/dynamodb_extractor.py

```diff
@@ -0,0 +1,199 @@
+    list: ("LIST", recursive_dynamo_format_converter)
+}
+
+
+def convert_value_list_to_dynamo_format(value_list: List[Any]):
+    return_values = []
+    for value in value_list:
+        return_values.append(convert_value_to_dynamo_format(value))
+    return return_values
+
+
+class DynamoConditional:
+    @classmethod
+    def from_conditional_data(cls, conditional_data: Dict[str, Any]):
+        return cls(**conditional_data)
+    
+    def __init__(self, AttributeValueList: List[Any], ComparisonOperator: str):
+        self.AttributeValueList: List[DynamoObject] = convert_value_list_to_dynamo_format(AttributeValueList)
+        self.ComparisonOperator = ComparisonOperator
+
+    def generate_filter_format(self):
+        return {'AttributeValueList': [dynamo_object.as_dynamo_representation() for dynamo_object in self.AttributeValueList], 'ComparisonOperator': self.ComparisonOperator}
+
+
+class DynamoFilterConverter:
+    @classmethod
+    def from_file_data(cls, scan_filter: Dict[str, Any]):
+        return cls({attribute_name: DynamoConditional.from_conditional_data(conditional_data) for attribute_name, conditional_data in scan_filter.items()})
+    
+    def __init__(self, conditionals: Dict[str, DynamoConditional]):
+        self.conditionals = conditionals
+
+    def generate_filter_converter(self):
+        return {attribute_name: conditional.generate_filter_format() for attribute_name, conditional in self.conditionals.items()}
+    
+
+class DynamoDBExtractor(Extractor):
+    @classmethod
+    def from_file_data(
+        cls,
+        table: str,
+        scan_filter: Dict[str, Any] = {},
+        limit: int = 100,
+        **aws_client_args,
+    ):
+        print(aws_client_args)
+        client = AwsClientFactory(**aws_client_args).make_client("dynamodb")
+        return cls(
+            table=table,
+            limit=limit,
+            scan_filter=scan_filter,
+            client=client,
+        )
+
+    def __init__(
+        self,
+        table,
+        limit: int,
+        scan_filter: Dict[str, Any],
+        client,
+    ) -> None:
+        self.table = table
+        self.client = client
+        self.scan_filter = DynamoFilterConverter.from_file_data(scan_filter).generate_filter_converter()
+        self.limit = limit
+        self.logger = getLogger(self.__class__.__name__)
+
+    def scan_table(self):
+        items = []
+        should_continue = True
+        parameters = { "TableName": self.table, "Limit": self.limit}
+        if self.scan_filter:
+            parameters.update({"ScanFilter": self.scan_filter})
+
+        while (should_continue):
+            response: dict = self.client.scan(
+                **parameters
+            )
+            items += response["Items"]
+            last_evaluated_key = response.get("LastEvaluatedKey", None)
+            if last_evaluated_key is not None:
+                should_continue = True
+                parameters.update({"ExclusiveStartKey": last_evaluated_key})
```

Comment:
```suggestion
                parameters["ExclusiveStartKey"] = last_evaluated_key)
```

No need to allocate a dict here.